pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/OnHeapStringDictionary.java

```diff
@@ -55,6 +58,7 @@ public OnHeapStringDictionary(PinotDataBuffer dataBuffer, int length, int numByt
       String unpaddedString = getUnpaddedString(i, buffer);
       _unpaddedStrings[i] = unpaddedString;
       _unPaddedStringToIdMap.put(unpaddedString, i);
+      _unpaddedBytes[i] = getUnpaddedBytes(i, buffer);
```

Comment:
Only read the unpadded bytes, and directly create unpaddedString from the unpadded bytes for better performance

Reply:
The performance will not be good because we have do encode every time if we want to read unpadded string? Or you are talking about space efficiency?

Reply:
I'm talking about the bytes/string reading during the construction
```suggestion
     byte[] unpaddedBytes = getUnpaddedBytes(i, buffer);
     String unpaddedString = new String(unpaddedBytes, UTF_8);
     ...
```